plugins/compile.sh

```diff
@@ -7,18 +7,33 @@
 #
 # This file is part of AMX Mod X.
 
-# new code contributed by \malex\
+# to compile all source files in the directory
+#  execute the script without any parameters
+# to compile specific or multiple source files
+#  execute the script with parameters set as source files' name
+#  e.g ./compile.sh admin.sma admincmd.sma adminchat.sma
 
 test -e compiled || mkdir compiled
 rm -f temp.txt
 
-for sourcefile in *.sma
-do
-        amxxfile="`echo $sourcefile | sed -e 's/\.sma$/.amxx/'`"
-        echo -n "Compiling $sourcefile ..."
-        ./amxxpc $sourcefile -ocompiled/$amxxfile >> temp.txt
-        echo "done"
-done
+if [ $# != 0 ]
+	then
+		for src in "$@"
```

Comment:
Should `src` be validated that it ends with `.sma`?

Reply:
Fixed it.